packages/denylist/src/denylist.js

```diff
@@ -0,0 +1,36 @@
+/* eslint-env serviceworker, browser */
+
+import { normalizeCid } from './utils/cid'
+import { getFromDenyList } from './utils/denylist'
+import { JSONResponse } from '@web3-storage/worker-utils/response'
+
+/**
+ * Returns badbits denylist results.
+ * @param {Request} request
+ * @param {import('./env').Env} env
+ */
+export const denylistGet = async function (request, env) {
+  const cid = (new URL(request.url)).pathname?.substring(1)
```

Comment:
You should be able to do here, dropping the query param kind of route that I commented in index.js file

```suggestion
  const cid = request.params.cid
```

Reply:
The type request is not Web Request, but itty router request https://github.com/kwhitley/itty-router#example

So, types need to be updated to receive `@param {import('itty-router')} request` instead